.github/workflows/ci.yml

```diff
@@ -428,6 +461,84 @@ jobs:
           export PATH="$PWD/bin:$PWD/dist/bin:$PATH"
           dist-tests/${{ matrix.suite }}
 
+  coverage:
+    name: "Run integration tests with coverage reporting"
+    needs: build
+    runs-on: ${{ matrix.os }}
+    if: github.event_name == 'pull_request'
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
Is this step needed?  Seems like you are downloading the various pre-built artifacts below so are sources still needed?

Reply:
Unfortunately HPC needs the sources to generate the HTML coverage report.